pkg/agent/controller/egress/egress_controller.go

```diff
@@ -566,24 +567,63 @@ func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP s
 	if egressIP != "" {
 		isLocal = c.localIPDetector.IsLocalIP(egressIP)
 	}
+
+	desiredStatus := &crdv1b1.EgressStatus{}
+	if isLocal {
+		desiredStatus.EgressNode = c.nodeName
+		desiredStatus.EgressIP = egressIP
+		if isEgressSchedulable(egress) {
+			desiredStatus.Conditions = []crdv1b1.EgressCondition{
+				{
+					Type:               crdv1b1.IPAssigned,
+					Status:             v1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "Assigned",
+					Message:            "EgressIP is successfully assigned to EgressNode",
+				},
+			}
+		}
+	} else if egressIP == "" {
+		// Select one Node to update false status among all Nodes.
+		nodeToUpdateStatus, err := c.cluster.SelectNodeForIP(egress.Spec.EgressIP, "")
+		if err != nil {
+			return err
+		}
+		// Skip if the Node is not the selected one.
+		if nodeToUpdateStatus != c.nodeName {
+			return nil
+		}
+		desiredStatus.EgressNode = ""
+		desiredStatus.EgressIP = ""
+		if isEgressSchedulable(egress) {
+			desiredStatus.Conditions = []crdv1b1.EgressCondition{
+				{
+					Type:               crdv1b1.IPAssigned,
+					Status:             v1.ConditionFalse,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "NoAvailableNode",
+					Message:            "No available Node can be elected as EgressNode",
+				},
+			}
+		}
+	} else {
+		return nil
+	}
+
 	toUpdate := egress.DeepCopy()
 	var updateErr, getErr error
 	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		if isLocal {
-			// Do nothing if the current EgressNode in status is already this Node.
-			if toUpdate.Status.EgressNode == c.nodeName && toUpdate.Status.EgressIP == egressIP {
-				return nil
-			}
-			toUpdate.Status.EgressNode = c.nodeName
-			toUpdate.Status.EgressIP = egressIP
-		} else {
-			// Do nothing if the current EgressNode in status is not this Node.
-			if toUpdate.Status.EgressNode != c.nodeName {
-				return nil
+		needsToBeUpdated := compareEgressStatus(toUpdate.Status, *desiredStatus)
+		if !needsToBeUpdated {
+			return nil
+		}
```

Comment:
Like the handling in pkg/controller/egress/controller.go, it could be (after you change the return value of `compareEgressStatus` to true when they equal.
```
if compareEgressStatus(toUpdate.Status, *desiredStatus) {
        return nil
}
```